js/runtime.js

```diff
@@ -893,17 +907,23 @@
             clearTo: new util.Method()
                 .when(['string'], function(clr){ // unfortunately colors are still strings
                     var r = canvasRect();
-                    resetCanvas();
-                    getContext().fillStyle = clr;
-                    getContext().fillRect(r.x, r.y, r.width, r.height);
+                    var c = getContext();
```

Comment:
we were calling clearContext before, which is overkill. What we need to do is save and restore the context around clearing the stage.